### ISSUE TITLE
Analysis incorrectly wants kotlin-reflect to be removed or changed to runtime-only

### ISSUE THREAD

**Issue by Mahoney** [autonomousapps/dependency-analysis-gradle-plugin#976]
**Build scan link**
https://scans.gradle.com/s/zd5eikniyguvg

**Plugin version**
1.24.0

**Gradle version**
8.3

**(Optional) Android Gradle Plugin (AGP) version**
N/A

**`reason` output for bugs relating to incorrect advice**
```
> Task :lidalia-uri:reason

----------------------------------------
You asked about the dependency 'org.jetbrains.kotlin:kotlin-reflect:1.9.10'.
You have been advised to change this dependency to 'testRuntimeOnly' from 'testImplementation'.
----------------------------------------

There is no path from :lidalia-uri to org.jetbrains.kotlin:kotlin-reflect:1.9.10 for compileClasspath


There is no path from :lidalia-uri to org.jetbrains.kotlin:kotlin-reflect:1.9.10 for runtimeClasspath


Shortest path from :lidalia-uri to org.jetbrains.kotlin:kotlin-reflect:1.9.10 for testCompileClasspath:
:lidalia-uri
\--- org.jetbrains.kotlin:kotlin-reflect:1.9.10

Shortest path from :lidalia-uri to org.jetbrains.kotlin:kotlin-reflect:1.9.10 for testRuntimeClasspath:
:lidalia-uri
\--- org.jetbrains.kotlin:kotlin-reflect:1.9.10

Source: main
------------
(no usages)

Source: test
------------
* Provides 1 service loader: kotlin.reflect.jvm.internal.impl.serialization.deserialization.builtins.BuiltInsLoaderImpl (implies testRuntimeOnly).
```

**Describe the bug**
The plugin suggests that kotlin-reflect should be removed from the dependencies or declared as runtime only.

However, if I do either of those then compilation fails, because I am using `Foo::class.supertypes` which can only compile with kotlin-reflect on the compile time classpath.

**To Reproduce**
1. Create a JVM kotlin project with the build health plugin
2. Add a dependency `implementation(kotlin("reflect"))`
3. Use `String::class.supertypes` somewhere in the main source
4. Run `./gradlew buildHealth`
5. Observe that the plugin says that `kotlin-reflect` should be removed or made runtime-only
6. Observe that if you remove `kotlin-reflect` or make it runtime only the project fails to compile

**Expected behavior**
No warnings from the plugin

**Additional context**
N/A

**Comment by Mahoney** [comment]
Workaround is of course:
```
dependencyAnalysis {
  issues {
    all {
      onAny {
        exclude("org.jetbrains.kotlin:kotlin-reflect")
      }
    }
  }
}
```

**Comment by autonomousapps** [comment]
Thanks for the issue. I wonder if this relates to https://github.com/autonomousapps/dependency-analysis-gradle-plugin/issues/987 ?

**Comment by Mahoney** [comment]
Yup, I can confirm it's fixed in v1.26.0
